### PR TITLE
Update bsp

### PR DESCRIPTION
**Pull request by daniel-dsouza**
This should allow you to use the 32.2 versions of the CTI BSP. still needs testing.

### CODE REVIEW

File: docker/cti/Makefile [hunk 1]
```diff
@@ -1,14 +1,14 @@
 #!make
 
-cti-all-deps: cti-32.1-jax-deps cti-32.1-tx2-deps cti-32.1-tx2i-deps
```
Comment (idavis) [inline docker/cti/Makefile]: The changes in this file will need to be additive like the others to maintain 32.1 builds.

**Comment by daniel-dsouza** [comment]
another make question for you @idavis 

I seem to be getting this error
```
 $ make l4t-32.2-tx2
make -C /home/daniel/source/jetson-containers/docker/l4t 32.2-tx2
make[1]: Entering directory '/home/daniel/source/jetson-containers/docker/l4t'
docker build --squash \
                                --build-arg VERSION_ID="bionic-20190612" \
                                --build-arg DEPENDENCIES_IMAGE=l4t:32.2-tx2-jetpack-4.2.1-deps \
                                -t l4t:32.2-tx2 \
                                -f /home/daniel/source/jetson-containers/docker/l4t/32.2/tx2/Dockerfile \
                                .
unable to resolve docker endpoint: context "." does not exist
Makefile:21: recipe for target '32.2-tx2' failed
make[1]: *** [32.2-tx2] Error 1
make[1]: Leaving directory '/home/daniel/source/jetson-containers/docker/l4t'
Makefile:39: recipe for target 'l4t-32.2-tx2' failed
make: *** [l4t-32.2-tx2] Error 2
```
although running the same build command inside `~/source/jetson-containers/docker/l4t` works just fine. Did I miss something in the makefile?

**Comment by idavis** [comment]
@daniel-dsouza I've heard this issue from one other user. There is something environmental AFAICT. I just ran the build locally and it worked fine:

```bash
idavis@__:~/jetson-containers$ make l4t-32.2-tx2
make -C /home/<user>/jetson-containers/docker/l4t 32.2-tx2
make[1]: Entering directory '/home/<user>/jetson-containers/docker/l4t'
docker build --squash \
                                --build-arg VERSION_ID="bionic-20190612" \
                                --build-arg DEPENDENCIES_IMAGE=l4t:32.2-tx2-jetpack-4.2.1-deps \
                                -t l4t:32.2-tx2 \
                                -f /home/<user>/jetson-containers/docker/l4t/32.2/tx2/Dockerfile \
                                .
Sending build context to Docker daemon  37.89kB
Step 1/21 : ARG VERSION_ID
Step 2/21 : ARG DEPENDENCIES_IMAGE
Step 3/21 : ARG IMAGE_NAME
Step 4/21 : FROM ${DEPENDENCIES_IMAGE} as dependencies
 ---> 285ba7b2d8a9
```

**Comment by daniel-dsouza** [comment]
@idavis which version of docker do you have?
> Docker version 19.03.2, build 6a30dfc

**Comment by idavis** [comment]
```
Client:
 Version:           18.09.7
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        2d0083d
 Built:             Thu Jun 27 17:56:23 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.7
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       2d0083d
  Built:            Thu Jun 27 17:23:02 2019
  OS/Arch:          linux/amd64
  Experimental:     true
```

**Comment by daniel-dsouza** [comment]
hmm. given that docker has changed the way context work in 19.03, this might be related to that...

**Comment by idavis** [comment]
Are you using 18.10 or manually updating? That shouldn't be available on 18.04 LTS.

**Comment by daniel-dsouza** [comment]
It's in the repos for me

```
 $ apt-cache policy docker-ce
docker-ce:
  Installed: 5:19.03.2~3-0~ubuntu-bionic
  Candidate: 5:19.03.2~3-0~ubuntu-bionic
  Version table:
 *** 5:19.03.2~3-0~ubuntu-bionic 500
        500 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
        100 /var/lib/dpkg/status
...and many more.
```

**Comment by idavis** [comment]
You are right. I haven't updated. I'll try to update and give it a run.

**Comment by idavis** [comment]
Well, we have a repro...

```
idavis@__:~/jetson-containers$ docker version
Client:
 Version:           18.09.7
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        2d0083d
 Built:             Thu Jun 27 17:56:23 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.7
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       2d0083d
  Built:            Thu Jun 27 17:23:02 2019
  OS/Arch:          linux/amd64
  Experimental:     true
idavis@__:~/jetson-containers$ make l4t-32.2-tx2
make -C /home/<user>/jetson-containers/docker/l4t 32.2-tx2
make[1]: Entering directory '/home/<user>/jetson-containers/docker/l4t'
docker build --squash \
                                --build-arg VERSION_ID="bionic-20190612" \
                                --build-arg DEPENDENCIES_IMAGE=l4t:32.2-tx2-jetpack-4.2.1-deps \
                                -t l4t:32.2-tx2 \
                                -f /home/<user>/jetson-containers/docker/l4t/32.2/tx2/Dockerfile \
                                .
unable to resolve docker endpoint: context "." does not exist
Makefile:21: recipe for target '32.2-tx2' failed
make[1]: *** [32.2-tx2] Error 1
make[1]: Leaving directory '/home/<user>/jetson-containers/docker/l4t'
Makefile:39: recipe for target 'l4t-32.2-tx2' failed
make: *** [l4t-32.2-tx2] Error 2
```

**Comment by idavis** [comment]
@daniel-dsouza Do you have any info on the change? I'm not seeing anything in the release notes that is shedding light.

**Comment by idavis** [comment]
@daniel-dsouza Also, if I copy the exact command that was run and execute it, the build works fine...

**Comment by daniel-dsouza** [comment]
> @daniel-dsouza Do you have any info on the change? I'm not seeing anything in the release notes that is shedding light.

I found the [relevant discussion](https://github.com/docker/cli/pull/1501) about contexts, but I didn't see anything relating to the issue.

**Comment by idavis** [comment]
I'll look at that discussion more. I've narrowed it down to the CLI. I can run old and new engines, but the 19.x cli breaks us.

**Comment by idavis** [comment]
@daniel-dsouza I have it figured out.

https://github.com/simonferquel/cli/blob/591385a1d0431b6ec7b02e26b0768f7b96d31df9/cli/command/cli.go#L455

In my makefile I created a var called DOCKER_CONTEXT which gets expanded on the command line. They now have their own with the same name and they check it before they resolve `.`, but it requires a real directory name as they won't expand the `.` when it's part of the env var.

I'm trying to figure out how to get around this.

**Comment by idavis** [comment]
@daniel-dsouza BTW, that discussion led to me figure it out as I was reading through the diff. Thank you very much.

**Comment by idavis** [comment]
@daniel-dsouza I need to test out https://github.com/idavis/jetson-containers/pull/20/files to make sure everything still works. You'll want to make the appropriate changes to your branch as well.

**Pull request merged**